src/Microsoft.OpenApi/Models/OpenApiComponents.cs

```diff
@@ -19,60 +20,60 @@ public class OpenApiComponents : IOpenApiSerializable, IOpenApiExtensible
         /// <summary>
         /// An object to hold reusable <see cref="JsonSchema"/> Objects.
         /// </summary>
-        public IDictionary<string, JsonSchema> Schemas { get; set; } = new Dictionary<string, JsonSchema>();
+        public IDictionary<string, JsonSchema>? Schemas { get; set; } = new Dictionary<string, JsonSchema>();
```

Comment:
for the properties which have a default value, should we instead avoid making them nullable? Doing so will require null propagation/checks all over (for consumers as well) and degrade the experience.

For the ones that don't, would it be worth adding a default value?

Reply:
I tried, but if you look at the constructors, they still allow null values ​​to be set. And since I'm not changing the behavior, I made it null.
Also it breaks tests in some scenarios because there is a difference between null and empty value in yaml/json files

Reply:
Do you mean the copy constructors? is that because of the null propagration?

Reply:
Yes. I think we need to change the copy constructors so that they do not allow null right away.
But still not sure which of these properties can be null because as I wrote above here the difference is in json/yaml
Perhaps only those marked REQUIRED in the comments should be not null?

Reply:
yes, I think that makes more sense, both from a standard implementation perspective and from a developer experience perspective.

Reply:
I believe @irvinesunday had already started some work on enabling null reference types https://github.com/microsoft/OpenAPI.NET/issues/1202

Reply:
Judging by the pr it looks abandoned, there's a lot more work to be done here